### PR TITLE
Fix edge distance calculation in duplication check of edge depth refinement

### PR DESCRIPTION
**Pull request by mmurooka**
This is very complicated bug...
Before PR, the distance of edges are calculated from the distance of nearest points in lines (line has no limitation of length). In that case, two nearly-parallel but little-tilting lines are sometimes calculated to be very near even if they are not.
This PR considers the length of edges, so this bug is fixed.

- Before PR
before_pr_edited
- After PR
after_pr_edited

### CODE REVIEW

File: jsk_recognition_utils/include/jsk_recognition_utils/geo/segment.h [hunk 1]
```diff
@@ -53,6 +53,7 @@ namespace jsk_recognition_utils
     virtual double distance(const Eigen::Vector3f& point) const;
     virtual double distance(const Eigen::Vector3f& point, Eigen::Vector3f& foot_point) const;
     virtual bool intersect(Plane& plane, Eigen::Vector3f& point) const;
+    virtual void midpoint(Eigen::Vector3f& output) const;
```
Comment (wkentaro) [inline jsk_recognition_utils/include/jsk_recognition_utils/geo/segment.h]: LGTM, but maybe `output` -> `midpoint` is better?

**Comment by mmurooka** [comment]
Thank you for review and merge.

File: jsk_recognition_utils/include/jsk_recognition_utils/geo/segment.h (hunk 1 above)
Comment (k-okada) [inline jsk_recognition_utils/include/jsk_recognition_utils/geo/segment.h]: @mmurooka forget to include this change, please send as another pR

Comment (mmurooka) [inline jsk_recognition_utils/include/jsk_recognition_utils/geo/segment.h]: fixed at https://github.com/jsk-ros-pkg/jsk_recognition/pull/2013
